### PR TITLE
feat(core/algorithms): highlight assertions in algorithms

### PR DESCRIPTION
**Pull request by Swapnilr1**
This is a preliminary pull request to solve issue #1762.

### CODE REVIEW

**Comment by marcoscaceres** [comment]
Oh, no problem. Happy to guide you. We use Jasmine as our testing framework:
https://jasmine.github.io/pages/getting_started.html

Give that a quick read. It's fairly simple. 

For your tests, you should be able to copy and paste this file:
https://github.com/w3c/respec/blob/develop/tests/spec/core/linter-rules/check-punctuation.js

And then just modify it to test your code. 

To run the tests, you can do:

```
npm start
```

And go to http://localhost:5000/tests/SpecRunner.html

In the test file, if you change  the `describe()` function to `fdescribe()`, that will "focus" the test; meaning it will only run those tests instead of the whole test suite. Will save you a bunch of time/pain :)  

Give it a go and give us a shout if you get stuck!

**Comment by Swapnilr1** [comment]
@marcoscaceres From what I have understood, I should make a file `algorithms-spec.js` in `tests/spec/core` and add the path to `testFiles.json` so that it gets added to the tests.
I have also understood the fundamentals of how tests work. In the test, I should write a sample document with algorithm list containing an assert statement.  
Then, at the end of the test I should have an `expect` statement to see if the class `assert` was added where required or not.

Problem is I can't figure out what this block is doing:
`
const results = await rule.lint(config, doc);
const [result] = results;
`
Supposedly, it is running the linter on the document just created above it. But how do I call my function from `algorithms.js` ? I mean where was the linter defined is not clear to me. 
I am sorry if this is too trivial to ask here.

**Comment by marcoscaceres** [comment]
oh whoops! Really sorry @Swapnilr1, I pointed you to the wrong test file! Maybe use this one instead:
https://github.com/w3c/respec/blob/develop/tests/spec/core/informative-spec.js

Copying the above code will give you a `doc` that has run your code ([line 11](https://github.com/w3c/respec/blob/develop/tests/spec/core/informative-spec.js#L11)). Then you can check the result by calling `doc.querySelector()` and querying for things that your code changed.

File: tests/spec/core/algorithms-spec.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  afterAll(flushIframes);
+  it("should add assert class to Assertions in algorithms", async () => {
+    const ops = {
+      config: makeBasicConfig(),
+      body:
+        makeDefaultBody() +
+        `<section>
+            <ol class='algorithm'>
+            <li>This one shouldn't be modified.</li>
+            <li>Assert: This one should have assert class added to it.</li>
```
Comment (marcoscaceres) [inline tests/spec/core/algorithms-spec.js]: This is looking good, but let's add a few more test cases... what happens if you have:
```HTML
  <li><!-- whitespace -->
     Assert: test
  <li>
```

File: src/core/algorithms.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+/**
+Currently used only for adding 'assert' class to algorithm lists  
+*/
+
+export const name = "core/algorithms";
+
+export async function run() {
+  const elements = Array.from(document.querySelectorAll("ol.algorithm li"));
+  elements
+    .filter(li => li.innerText.startsWith("Assert: "))
```
[the remark this answers is not in the document]
Comment (sidvishnoi) [reply src/core/algorithms.js]: Update: Might need to fall back to `trimLeft` or `trim` as Edge 42 doesn't support `trimStart`

[the remark this answers is not in the document]
Comment (marcoscaceres) [reply src/core/algorithms.js]: ...or maybe RegExp it?

[the remark this answers is not in the document]
Comment (sidvishnoi) [reply src/core/algorithms.js]: Something like: `/^\s*Assert\:/`

**Comment by Swapnilr1** [comment]
@marcoscaceres @sidvishnoi Thank you very much for the suggested changes. My bad, I missed removing the whitespaces when comparing. I have now incorporated the changes suggested.

**Comment by Swapnilr1** [comment]
@marcoscaceres Thank you very much. Would love to work on another issue.

**Comment by marcoscaceres** [comment]
Please feel free to grab any issue you find interesting... in particular: https://github.com/w3c/respec/issues/2005

**Pull request merged**